src/commons/temporal.h

```diff
@@ -27,4 +29,86 @@
 	*/
 	char *temporal_get_string_time(const char* format);
 
+	/**
+	* @NAME: t_state
+	* @DESC: Estado de una variable temporal.
+	*/
+	typedef enum {
+		STOPPED,
+		RUNNING
+	} t_state;
+
+	/**
+	* @NAME: t_temporal
+	* @DESC: Estructura de una Variable temporal.
+	*/
+	typedef struct {
+		struct timespec* current;
+		int64_t elapsed_ms;
+		t_state state;
+	} t_temporal;
+
+	/**
+	* @NAME: temporal_create
+	* @DESC: Crea una variable temporal.
+	*/
+	t_temporal* temporal_create(void);
+
+	/**
+	* @NAME: temporal_destroy
+	* @DESC: Destruye una variable temporal.
+	* @PARAMS: 
+	*		temporal - Variable temporal a destruir.
+	*/
+	void temporal_destroy(t_temporal* temporal);
+
+	/**
+	* @NAME: temporal_gettime
+	* @DESC: Retorna el tiempo transcurrido mientras el temporal estuvo activo en milisegundos.
+	* @PARAMS:
+	*		temporal - Variable temporal.
+	*/
+	int64_t temporal_gettime(t_temporal* temporal);
+	
+	/**
+	* @NAME: temporal_gettime_since_running
+	* @DESC: Retorna el tiempo transcurrido en el último período en el cual el temporal estuvo activo, en milisegundos.
+			 si el temporal fue detenido y reanudado, sólo retornará el tiempo transcurrido desde el último reanudado.
+			 Ejemplo:
+			 	t_temporal temporal = temporal_create();
+				sleep(2)
+				temporal_stop(temporal);
+				temporal_resume(temporal);
+				sleep(1);
+				temporal_gettime_since_running(temporal); => 1000 (Último período)
+	* @PARAMS:
+	*		temporal - Variable temporal.
+	*/
+	int64_t temporal_gettime_since_running(t_temporal* temporal);
+
+	/**
+	* @NAME: temporal_stop
+	* @DESC: "Frena" el timer de una variable temporal. Deja de sumar tiempo.
```

Comment:
["timer" (temporizador) != "stopwatch" (cronómetro)](https://en.wikipedia.org/wiki/Timer#:~:text=The%20word%20%22timer%22%20is%20usually,from%20zero%2C%20are%20called%20stopwatches.)
```suggestion
	* @DESC: Detiene el cronómetro de una variable temporal.
```